pkg/segment/writer/segstore.go

```diff
@@ -124,6 +130,22 @@ func NewSegStore(orgId uint64) *SegStore {
 	return segstore
 }
 
+func (ss *SegStore) GetNewBitset(bsSize uint) *bitset.BitSet {
+	if ss.bsPoolCurrIdx >= ss.bsPoolCount {
+		newCount := BS_INITIAL_SIZE
+		ss.bsPool = toputils.ResizeSlice(ss.bsPool, int(newCount+ss.bsPoolCount))
+		for i := uint32(0); i < newCount; i++ {
+			newBs := bitset.New(bsSize)
+			ss.bsPool[ss.bsPoolCount+i] = newBs
+		}
```

Comment:
nit: I've seen this same pattern recently (I think on of Mani's PRs), so we might want to make a new util like:
```
func ResizeSliceWithDefault[T any](slice []T, size int, constructor func() T)
```
(You don't need that change in this PR if you want)

Reply:
created a issue for this: https://github.com/siglens/siglens/issues/1525